### PR TITLE
chore: change to ubuntu base image for docker image

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Description

closes FL-1530

## Trivy scan

12 middle and 8 low

## Checklist

- [ ] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by guilhem-barthes — APPROVED** [review]
Thank you so much 🙏

File: docker/substra-backend/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 
-ARG DEBIAN_FRONTEND=noninteractive
-
-RUN apt-get update && apt-get upgrade -y && apt-get install -y --no-install-recommends \
-    build-essential \
-    && apt-get clean && rm -rf /var/lib/apt/lists/*
+RUN apt-get update -y \
+    && apt-get install --no-install-recommends -y --only-upgrade gnupg \
+    && apt-get install --no-install-recommends -y software-properties-common \
+    && apt-get --no-install-recommends -y install python3-pip gcc python3.11-dev python3-watchdog celery python-is-python3 python3.11-venv \
```
Comment (guilhem-barthes) [inline docker/substra-backend/Dockerfile:9]: we could merge these two lines
```suggestion
    && apt-get install --no-install-recommends -y software-properties-common \
    python3-pip gcc python3.11-dev python3-watchdog celery python-is-python3 python3.11-venv \
```

File: docker/substra-backend/Dockerfile [hunk 2]
```diff
@@ -32,8 +36,7 @@ FROM build AS arm64
 RUN apt-get update && apt-get install -y --no-install-recommends \
     # libpq is a required dependency to build psycopg2 from source (support for arm64 platform)
     libpq-dev
-RUN pip3 install --upgrade --no-cache-dir pip==24.0 \
-    && pip3 install --prefix /usr/local --no-cache-dir psycopg2==2.9.9
+RUN pip install --no-cache-dir psycopg2==2.9.9
```
Comment (guilhem-barthes) [inline docker/substra-backend/Dockerfile:39]: Is there a reason why we don't do it on line 19 / 20?

Comment (ThibaultFy) [reply docker/substra-backend/Dockerfile:39]: Yes, it's only meant to be installed for arm64, it is under `FROM build AS arm64`

**Pull request merged**